src/targets/gpu/preallocate_param.cpp

```diff
@@ -16,8 +16,9 @@ void preallocate_param::apply(module& p) const
     {
         if(ins->name() != "@param")
             continue;
-        std::string id = any_cast<builtin::param>(ins->get_operator()).parameter;
-        if(id != param)
+        std::string id        = any_cast<builtin::param>(ins->get_operator()).parameter;
+        std::string mod_param = p.name() + param;
+        if(id != mod_param)
```

Comment:
This should be:

```cpp
if (param != any_cast<builtin::param>(ins->get_operator()).parameter)
    continue;
std::string id = p.name() + ":" param;
```

This will match `scratch` parameters for all modules and give each one a unique id.

Reply:
fixed